assemble.py

```diff
@@ -89,7 +89,7 @@ def _resize_wic_file(self, increase_bytes: int, extra_space=0.2):
             'conv=notrunc', 'oflag=append', 'count=' + str(increase_k),
             'seek=' + str(wic_k))
 
-        parted_out = str(subprocess.check_output(['parted', self._path, 'print']))
+        parted_out = subprocess.check_output(['parted', self._path, 'print']).decode('ascii')
         if parted_out.find('Partition Table: gpt') != -1:
```

Comment:
str vs an error decoding feels like a weird compromise. my guess we need `.decode("utf-8")` or just `.decode()`. However, we can keep this even simpler and not bother decoding by doing:
```
        parted_out = subprocess.check_output(['parted', self._path, 'print'])
        if parted_out.find(b'Partition Table: gpt') != -1:
```

Reply:
True, but then I can't use split*() on the byte encoded data.  (Which was the reason behind this change)

Reply:
split works for both bytes and strings

Reply:
This was kind of a nice simplification, last commit changes:
- Drop str() call for `parted_out` which leaves it as byte data
- Adjust `find` and `split` to use to byte data
- Convert final `_last_part` to int for storage -- should break nicely if we every have an issue here

Reply:
And a few cleanups from that conversion to int.
Re-tested the code path for both GPT and msdos images.